table/block_based_table_reader.cc

```diff
@@ -214,10 +213,10 @@ class PartitionIndexReader : public IndexReader, public Cleanable {
   }
 
  private:
-  PartitionIndexReader(BlockBasedTable* table, const Comparator* comparator,
+  PartitionIndexReader(BlockBasedTable* table,
                        std::unique_ptr<Block>&& index_block, Statistics* stats,
                        const int level)
-      : IndexReader(comparator, stats),
+      : IndexReader(&table->rep_->internal_comparator, stats),
```

Comment:
@maysamyabandeh @yiwu-arbug 

Now the eviction of data blocks has an even stricter condition. Before the change, an index reader cannot outlive DB. Now it cannot outlive the table reader object, which can happen much more likely than the former case.

Although I can't see a problem so far, it still feels scaring to me.